### PR TITLE
Outdated ICV restrictions were removed;

### PR DESCRIPTION
**Pull request by lupustr3**
### This pullrequest changes

Removes outdated restrictions on ICV API, since ICV is in sync with IPP

### CODE REVIEW

**Comment by vpisarev** [comment]
it's fine with me! @alalek, what do you think?

**Comment by alalek** [comment]
@lupustr3 There are other places where IPPICV difference is used. Should we [remove](https://github.com/alalek/opencv/commit/57871d5f28d0969c6ddaf0f8f8d5d275ae5202c6) them also?

File: modules/core/src/stat.cpp [hunk 1]
```diff
@@ -2282,10 +2278,10 @@ static bool ipp_minMaxIdx( Mat &src, double* minVal, double* maxVal, int* minIdx
                 depth == CV_8S ? (ippiMinMaxIndxFuncC1)ippiMinMaxIndx_8s_C1R :
 #endif
                 depth == CV_16U ? (ippiMinMaxIndxFuncC1)ippiMinMaxIndx_16u_C1R :
-#if !((defined _MSC_VER && defined _M_IX86) || defined __i386__)
+#if IPP_DISABLE_BLOCK
```
Comment (alalek) [inline modules/core/src/stat.cpp]: We observed issues here on 32-bit builds. Tests for x64 builds were fine on AMD CPUs.

What is about this condition?
```
#if !((defined _MSC_VER && defined _M_IX86) || defined __i386__) || IPP_DISABLE_BLOCK
```

**Comment by lupustr3** [comment]
@alalek I missed one in modules/imgproc/test/test_houghLines.cpp, yes

Concerning other cases:
I do not want to remove HAVE_IPP_ICV_ONLY define completely, just in case.
And I think it is better for implementation detector to be able to tell difference between packages.

File: modules/core/src/stat.cpp (hunk 1 above)
Comment (lupustr3) [inline modules/core/src/stat.cpp]: Is there details on this issue, track id? I see id only for NAN problem.

**Comment by alalek** [comment]
:+1:

**Pull request merged**